gufe/settings/models.py

```diff
@@ -0,0 +1,174 @@
+"""
+Pydantic models used for storing settings.
+"""
+
+from datetime import date
+from typing import Literal, Union
+
+from pydantic import BaseModel, Extra, FilePath, PositiveFloat
+from typing_extensions import TypedDict
+
+
+class SettingsBaseModel(BaseModel):
+    """
+    Settings and modifications we want for all settings classes.
+    """
+
+    class Config:
+        extra = Extra.forbid
+        # Immutability in python is never strict.
+        # If developers are determined/stupid
+        # they can always modify a so-called "immutable" object.
+        # https://pydantic-docs.helpmanual.io/usage/models/#faux-immutability
+        allow_mutation = False
+        arbitrary_types_allowed = True
+        smart_union = True
```

Comment:
I'm not sure we have to worry about settings being mutable.  I think we'll store this as a "private" variable, then only return copies of it when it is accessed.

Reply:
I don't think we need to go to extra effort, but I don't want to have to worry about the validation code having to re-check things, so I think I would prefer for this to act immutable.